OtripleS.Web.Api/Controllers/StudentGuardiansController.cs

```diff
@@ -0,0 +1,196 @@
+            }
+            catch (StudentGuardianValidationException studentGuarduanValidationException)
+                when (studentGuarduanValidationException.InnerException is AlreadyExistsStudentGuardianException)
+            {
+                string innerMessage = GetInnerMessage(studentGuarduanValidationException);
+
+                return Conflict(innerMessage);
+            }
+            catch (StudentGuardianValidationException studentGuarduanValidationException)
+            {
+                string innerMessage = GetInnerMessage(studentGuarduanValidationException);
+
+                return BadRequest(innerMessage);
+            }
+            catch (StudentGuardianDependencyException studentGuardianDependentcyException)
+            {
+                return Problem(studentGuardianDependentcyException.Message);
+            }
+            catch (StudentGuardianServiceException studentGuardianServiceException)
+            {
+                return Problem(studentGuardianServiceException.Message);
+            }
+        }
+
+        [HttpGet]
+        public ActionResult<IQueryable<StudentGuardian>> GetAllStudentGuardians()
+        {
+            try
+            {
+                IQueryable<StudentGuardian> StoragestudentGuardians = studentGuardianService.RetrieveAllStudentGuardians();
+
+                return Ok(StoragestudentGuardians);
+            }
+            catch (StudentGuardianDependencyException studentGuardianDependentcyException)
+            {
+                return Problem(studentGuardianDependentcyException.Message);
+            }
+            catch (StudentGuardianServiceException studentGuardianServiceException)
+            {
+                return Problem(studentGuardianServiceException.Message);
+            }
+        }
+
+        [HttpGet]
+        public async ValueTask<ActionResult<StudentGuardian>> GetStudentGuardianByIdAsync(Guid studentId, Guid guardianId)
+        {
+            try
+            {
+                StudentGuardian storageStudentGuardian =
+                    await this.studentGuardianService.RetrieveStudentGuardianByIdAsync(studentId, guardianId);
+
+                return Ok(storageStudentGuardian);
+            }
+            catch (StudentGuardianValidationException studentGuardianValidationException)
+                when (studentGuardianValidationException.InnerException is NotFoundStudentGuardianException)
+            {
+                string innerMessage = GetInnerMessage(studentGuardianValidationException);
+
+                return NotFound(innerMessage);
+            }
+            catch (StudentGuardianValidationException studentGuarduanValidationException)
+            {
+                string innerMessage = GetInnerMessage(studentGuarduanValidationException);
+
+                return BadRequest(innerMessage);
+            }
+            catch (StudentGuardianDependencyException studentGuardianDependentcyException)
```

Comment:
Spellcheck "studentGuardianDependentcyException" please

Reply:
@viralpandya  thank you so much, its fixed.